File: import_products.py
import json
import re
import psycopg2
from psycopg2.extras import execute_values
import os

def slugify(text):
    text = text.lower()
    text = re.sub(r'[^\w\s-]', '', text)
    text = re.sub(r'[-\s]+', '-', text).strip('-')
    return text[:120]

def import_products():
    # Use environment variables for connection
    dbname = os.environ.get('PGDATABASE')
    user = os.environ.get('PGUSER')
    password = os.environ.get('PGPASSWORD')
    host = os.environ.get('PGHOST')
    port = os.environ.get('PGPORT')

    conn = psycopg2.connect(
        dbname=dbname,
        user=user,
        password=password,
        host=host,
        port=port
    )
    cur = conn.cursor()

    try:
        with open('/tmp/products.json', 'r') as f:
            products = json.load(f)

        # Get unique brands
        brands = set()
        for p in products:
            if p.get('brand'):
                brands.add(p['brand'])

        # Upsert brands
        brand_map = {}
        for brand_name in brands:
            slug = slugify(brand_name)
            cur.execute("""
                INSERT INTO public.brands (name, slug)
                VALUES (%s, %s)
                ON CONFLICT (slug) DO UPDATE SET name = EXCLUDED.name
                RETURNING id;
            """, (brand_name, slug))
            brand_map[brand_name] = cur.fetchone()[0]

        # Prepare products data
        product_data = []
        for p in products:
            internal_code = p.get('internal_code')
            name = p.get('name', '')
            long_desc = p.get('long_description', '')
            short_desc = long_desc[:280] if long_desc else ''
            brand_name = p.get('brand')
            brand_id = brand_map.get(brand_name)
            wholesale_price = p.get('wholesale_price')
            price = wholesale_price if wholesale_price is not None else 0
            weight = p.get('weight_kg', 0)
            ncm = p.get('ncm')
            slug = slugify(f"{internal_code}-{name}")

            product_data.append((
                internal_code,
                internal_code, # sku
                name,
                long_desc,
                short_desc,
                brand_id,
                wholesale_price,
                price,
                weight,
                ncm,
                True, # is_active
                False, # wholesale_only
                1, # moq
                slug
            ))

        # Upsert products
        query = """
            INSERT INTO public.products (
                internal_code, sku, name, description, short_description,
                brand_id, wholesale_price, price, weight, ncm,
                is_active, wholesale_only, moq, slug
            )
            VALUES %s
            ON CONFLICT (internal_code) DO UPDATE SET
                sku = EXCLUDED.sku,
                name = EXCLUDED.name,
                description = EXCLUDED.description,
                short_description = EXCLUDED.short_description,
                brand_id = EXCLUDED.brand_id,
                wholesale_price = EXCLUDED.wholesale_price,
                price = EXCLUDED.price,
                weight = EXCLUDED.weight,
                ncm = EXCLUDED.ncm,
                is_active = EXCLUDED.is_active,
                wholesale_only = EXCLUDED.wholesale_only,
                moq = EXCLUDED.moq,
                slug = EXCLUDED.slug,
                updated_at = now();
        """
        execute_values(cur, query, product_data)
        
        conn.commit()
        print(f"Successfully imported {len(products)} products.")

    except Exception as e:
        conn.rollback()
        print(f"Error: {e}")
    finally:
        cur.close()
        conn.close()

if __name__ == "__main__":
    import_products()

File: import_products.js
import fs from 'fs';
import { createClient } from '@supabase/supabase-js';

const SUPABASE_URL = process.env.SUPABASE_URL;
const SUPABASE_SERVICE_ROLE_KEY = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!SUPABASE_URL || !SUPABASE_SERVICE_ROLE_KEY) {
  console.error("Missing environment variables SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY");
  process.exit(1);
}

const supabase = createClient(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY);

function slugify(text) {
  if (!text) return "";
  return text.toString().toLowerCase()
    .normalize("NFD")
    .replace(/[\u0300-\u036f]/g, "")
    .replace(/[^a-z0-9]+/g, "-")
    .replace(/-+/g, "-")
    .replace(/^-|-$/g, "")
    .slice(0, 120);
}

async function importProducts() {
  try {
    const productsPath = fs.existsSync('scripts/extracted/products.json') 
      ? 'scripts/extracted/products.json' 
      : 'products.json';
    
    if (!fs.existsSync(productsPath)) {
      console.error(`File not found: ${productsPath}`);
      process.exit(1);
    }

    const products = JSON.parse(fs.readFileSync(productsPath, 'utf8'));
    console.log(`Loading ${products.length} products...`);

    const brandNames = [...new Set(products.map(p => p.brand).filter(Boolean))];
    const brandMap = new Map();

    for (const brandName of brandNames) {
      const slug = slugify(brandName);
      const { data, error } = await supabase
        .from('brands')
        .upsert({ name: brandName, slug: slug, is_active: true }, { onConflict: 'slug' })
        .select('id')
        .single();
      
      if (error) {
        console.error(`Error upserting brand ${brandName}:`, error.message);
        continue;
      }
      brandMap.set(brandName, data.id);
    }

    let okCount = 0;
    for (let i = 0; i < products.length; i += 50) {
      const chunk = products.slice(i, i + 50);
      const payloads = chunk.map(p => {
        if (!p.internal_code) return null;
        const brandId = brandMap.get(p.brand);
        const name = p.name || "";
        const description = p.long_description || "";
        const short_description = description.slice(0, 280);
        const wholesale_price = typeof p.wholesale_price === 'number' ? p.wholesale_price : null;
        
        return {
          internal_code: p.internal_code,
          sku: p.internal_code,
          name: name,
          description: description,
          short_description: short_description,
          brand_id: brandId,
          wholesale_price: wholesale_price,
          price: wholesale_price || 0,
          weight: p.weight_kg || 0,
          ncm: p.ncm || null,
          is_active: true,
          wholesale_only: false,
          moq: 1,
          slug: slugify(`${p.internal_code}-${name}`),
          updated_at: new Date().toISOString()
        };
      }).filter(Boolean);

      const { error } = await supabase
        .from('products')
        .upsert(payloads, { onConflict: 'internal_code' });

      if (error) {
        console.error(`Error upserting chunk:`, error.message);
      } else {
        okCount += payloads.length;
        console.log(`Progress: ${okCount}/${products.length}`);
      }
    }

    console.log(`Successfully imported ${okCount} products.`);
    
    const { data: stats, error: statsErr } = await supabase.rpc('get_import_stats');
    if (!statsErr) {
       console.log('STATS:', JSON.stringify(stats));
    } else {
       // Fallback if RPC doesn't exist yet
       const { count: total } = await supabase.from('products').select('*', { count: 'exact', head: true });
       console.log('TOTAL PRODUCTS:', total);
    }

  } catch (err) {
    console.error('Error:', err);
  }
}

importProducts();
